### PR TITLE
GITLAB-787 Add MailQueue:browse scenario

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministrationQuery.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.apache.james.gatling.control
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import io.gatling.http.request.builder.HttpRequestBuilder
+
+import java.net.URL
+
+class JamesWebAdministrationQuery(val baseUrl: URL) {
```
Comment (mbaechler) [inline src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministrationQuery.scala:9]: I find that naming strange. Aren't methods in JamesWebAdministration queries too ?

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapQueueBrowseScenario.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  val users = new UserCreator(BaseJamesWebAdministrationUrl).createUsersWithInboxAndOutbox(UserCount)
+
+  val webAdmin = new JamesWebAdministrationQuery(BaseJamesWebAdministrationUrl)
+
+  val count = new java.util.concurrent.atomic.AtomicInteger(0)
+
+  val scn = scenario("JmapSendMessages")
+    .exec(CommonSteps.provisionSystemMailboxes(users))
+    .during(ScenarioDuration) {
+      exec(JmapMessages.sendMessagesRandomlyWithRetryAuthentication(users))
```
Comment (mbaechler) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapQueueBrowseScenario.scala]: I'd rather write : 
```
exec(
  randomSwitch(
    Map(99 -> JmapMessages.sendMessagesRandomlyWithRetryAuthentication(users),
            1 -> webAdmin.getMailQueueMails("spool"))))
```
WDYT ?

File: src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministrationQuery.scala (hunk 1 above)
Comment (blackheaven) [reply src/main/scala-2.11/org/apache/james/gatling/control/JamesWebAdministrationQuery.scala:9]: Yes, but they are unlogged/untracked queries, I asked myself the question.

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapQueueBrowseScenario.scala (hunk 2 above)
Comment (blackheaven) [reply src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapQueueBrowseScenario.scala]: Great idea, I like it

**Comment by chibenwa** [comment]
Could you present us the result of a run on top of RabbitMQ?

**Review by aduprat — APPROVED** [review]

**Comment by blackheaven** [comment]
> Could you present us the result of a run on top of RabbitMQ?

@rouazana told me that RabbitMQ doesn't works yet, but as soon as it does, I'll do it.

**Review by mbaechler — APPROVED** [review]

**Review by rouazana — APPROVED** [review]

**Pull request merged**